Update retry-policies to 0.5

musicinmybrain opened the pull request:
The `build_with_total_retry_duration_and_max_retries method` of `retry_policies::policies::ExponentialBackoffBuilder` was renamed to `build_with_total_retry_duration_and_limit_retries`. No code changes are required, but since `reqwest-retry` re-exports `retry-policies::policies` as `reqwest_retry::policies`, this is a SemVer-breaking change for retry-policies as well.

<!-- Please explain the changes you made -->

<!--
Please, make sure:
- you have read the contributing guidelines:
  https://github.com/TrueLayer/reqwest-middleware/blob/main/CONTRIBUTING.md
- you have formatted the code using rustfmt:
  https://github.com/rust-lang/rustfmt
- you have checked that all tests pass, by running `cargo test --all`
- you have updated the changelog (if needed):
  https://github.com/TrueLayer/reqwest-middleware/blob/main/CHANGELOG.md
-->

jayvdb commented:
Would be a good idea to update the `CHANGELOG.md` as part of this PR, and probably mention it is a breaking change there

musicinmybrain commented:
> Heads up, there's a new version of retry-policies @musicinmybrain you could grab for this

I think `retry-policies = "0.5"` is still fine. It permits the latest `retry-policies`, which is now 0.5.1.

baszalmstra commented:
Could this be merged and released? This is currently blocking us to update some other dependencies as well.

suryapandian reviewed (commented):
Thank you for this 💮 
could you update the changelog in `reqwest-retry` and remove the changelog changes in `reqwest-middleware`? LGTM otherwise

musicinmybrain commented:
> Thank you for this 💮 could you update the changelog in `reqwest-retry` and remove the changelog changes in `reqwest-middleware`? LGTM otherwise

Thanks for noticing that! I moved the changelog entry to the correct crate and rebased on `main`.
